CleverRaven/Cataclysm-DDA
# Fix `zinc_metal` stats (and cascade effects)

#### Summary
Bugfixes "Fix `zinc_metal` stats (and cascade effects)"

#### Purpose of change
Split off from #62936

The density of `zinc_metal` is way off. Fix it, and deal with cascading effects in recipes.

#### Describe the solution

Zinc has density 7.14g/cm³. That's 357g for 50ml volume, which divides into a stack of 200 charges, 1785mg per unit.
&nbsp;

Adjust the disassembly of 5L zinc ingot, applying a 5.5% lossiness -- as I work my way through the spinoffs of #62936, 5~6% loss rate that will be applied consistently to all 5L ingot disassembly.
&nbsp;

Migrated item zinc oxide with misleading id `chem_zinc` to `chem_zinc_oxide` to prevent confusion with actual elemental/metal zinc items. Drive-by adjustment to volume due to discrepancy in quoted density figure -- not expected to affect recipes, since those depend on the weight (which corresponds to 1/100 moles of chemical).
&nbsp;

**Edit:** adopted `coal_lump` and `charcoal` stats from #62180
~Fix weight of `coal_lump`, from 1.5g to 37.5g. There was a mistake in 5549b62 where divided by count (250) instead of stack size (10) leading to value off by a factor of ×25. This is required for sanity of the next step.~
~Check cascade effects of coal weight change; see [details below](#issuecomment-1411414456)~
&nbsp;

As a source of elemental carbon for chemistry, `coal_lump` treated as anthracite coal with 90% carbon content.  
For `charcoal`, carbon content of wood charcoal can vary drastically (sources give 50~95%). Picked a fairly generous value of ~80.56% for convenience, since that will allow `charcoal` and `coal_lump` to be used at parity.
&nbsp;

Fix recipe for smelting `zinc_metal` from ore `material_zincite` by reduction with carbon. 
- Quantities reworked on the basis of zinc content of zincite, to ensure balanced input and output.
  - Zincite composition: Zn<sub>0.9</sub>Mn<sup>2+</sup><sub>0.1</sub>O <small>[[source](https://www.webmineral.com/data/Zincite.shtml)]</small>
- Added hammer and chisel tool requirements to break ore and coal into smaller pieces before roasting; would not work if you just chucked a large chunk of ore in.
- Replace basic boil and contain requirements with crucible due to high temperatures. 
- Added fuel required for roasting. This is distinct from the coal being used as source of carbon reagent. Smelting roughly 1.5kg of mass in total, or 6 steel-chunk-equivalents.
&nbsp;

Fix recipe for further reduction of zinc oxide with carbon; this uses an extremely conservative approach that oversupplies carbon.
- takes a hint from [this](https://royalsocietypublishing.org/doi/10.1098/rsos.170710), "to achieve the complete reduction of ZnO, the amount of reducing agent was increased to two times the stoichiometric amount."
- Replace basic boil and contain requirements with crucible due to high temperatures. 
- Added fuel required for roasting. This is distinct from the coal being used as source of carbon reagent. Smelting roughly 500g of mass in total, or 2 steel-chunk-equivalents.
&nbsp;

Adjust recipe for grinding of `zinc_metal` into `chem_zinc_powder`; reworked to produce ~130g, on the same scale as aluminum powder grinding recipe (176g output) and maintaining roughly the same material inefficiency as before (was 1308g/1400g, 6.6% lossy). This only balances zinc input and output, other issues with this recipe are marked with TODO and deferred to be fixed as a package with other problems found with powdered metals in #62936
&nbsp;

Adjust battery recipes by scaling `zinc_metal` inputs to roughly preserve mass. However, proportion also needs to be preserved when comparing among those recipes, so it is a straightforward scaling by 4× rather than 3.922× followed by rounding. (This is naive multiplication only, I did not look into the scientific basis for these recipes.)
&nbsp;

This should cover all pertinent references to `zinc_metal` in `src/`, `data/json/` and `data/mods`.

#### Testing
Suggested checks:
- Double check my maths / chemistry derivations.
- Load in game and check
  - zinc metal has sane stats
  - 5L ingot disassembly recipe reports sane input/output in crafting menu
  - check zinc smelting recipe in crafting menu for sanity
  - check zinc powder grinding recipe in crafting menu for sanity
  - check battery recipes in crafting menu for changes applied correctly.
